src/devices/cpu/avr8/avr8.h

```diff
@@ -33,21 +63,20 @@ class avr8_device : public cpu_device
 	// public interfaces
 	virtual void update_interrupt(int source);
 
-	// register handling
-	void regs_w(offs_t offset, uint8_t data);
-	uint8_t regs_r(offs_t offset);
-	uint32_t m_shifted_pc;
-
 	// GPIO
-	template<uint8_t Port> auto gpio_out() { return m_gpio_out_cb[Port].bind(); }
-	template<uint8_t Port> auto gpio_in() { return m_gpio_in_cb[Port].bind(); }
+	template <gpio_t Port> auto gpio_out() { return m_gpio_out_cb[Port].bind(); }
+	template <gpio_t Port> auto gpio_in() { return m_gpio_in_cb[Port].bind(); }
+	template <int Port> uint8_t pin_r();
+	template <int Port> void port_w(uint8_t data);
+	template <int Port> uint8_t gpio_r();
 
 	// ADC
 	template<uint8_t Pin> auto adc_in() { return m_adc_in_cb[Pin].bind(); }
 
 protected:
-	avr8_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock, const device_type type, uint32_t address_mask, address_map_constructor internal_map, int32_t num_timers);
+	avr8_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock, const device_type type, uint32_t address_mask, address_map_constructor internal_map);
 
+	typedef void (avr8_device::*timer_func) ();
 	typedef void (avr8_device::*op_func) (uint16_t op);
```

Comment:
Would you get any performance gain from making `timer_func` a delegate rather than a member function pointer, and avoiding one conditional branch on invocation?

Reply:
Should be addressed now.